app/tests/test_rapid.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+from unittest.mock import patch
+from fastapi.testclient import TestClient
+from core.config import ENSEMBL_URL
+from main import app
+
+
+class TestRapid(unittest.TestCase):
+    def setUp(self):
+        self.client = TestClient(app)
+        self.api_prefix = ""
+        self.mock_rapid_api_url = "/rapid"
+        self.species_url_name = "Human_GCA_123.1"
+        self.species_url_name_refseq = "DogRefSeq_GCA_123.1rs"
+
+        self.mock_genome_id_response1 = {
+            "genome_uuid": "genome_uuid1",
+            "release_version": 110,
+            "genome_tag": "",
+        }
+
+        self.mock_genome_id_response2 = {
+            "genome_uuid": "genome_uuid2",
+            "release_version": 110,
+            "genome_tag": "xyz",
+        }
+
+        self.mock_ncbi_accession = "GCF_000001405.40"
+
+        self.mock_resolved_url = {
+            "genome1": f"{ENSEMBL_URL}/species/genome_uuid1",
+            "genome2": f"{ENSEMBL_URL}/species/xyz",
+        }
+
+    @patch("api.resources.rapid_view.get_genome_id_from_assembly_accession_id")
+    @patch("api.resources.rapid_view.format_assembly_accession")
+    def test_rapid_species_home_success(
```

Comment:
With the way that you're handling the exceptions, could you please add a few more tests for 404, 500, using wrong protocols like POST and ...(anything that can trigger some of the exceptions). You can use AI to help with writing those tests too.